js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx

```diff
@@ -512,91 +604,111 @@ const DeleteConcurrencyLimitDialog = ({
+            text="Free all concurrency slots for run"
+            onClick={async () => {
+              const resp = await freeSlots({variables: {runId: pendingStep.runId}});
+              if (resp.data?.freeConcurrencySlots) {
+                onUpdate();
+                await showSharedToaster({
+                  intent: 'success',
+                  icon: 'copy_to_clipboard_done',
+                  message: 'Freed concurrency slots',
+                });
+              }
+            }}
+          />
+        </Menu>
+      }
+      position="bottom-right"
+    >
+      <Button rightIcon={<Icon name="expand_more" />}>Actions</Button>
+    </Popover>
+  );
+};
+
+const ConcurrencyStepsDialog = ({
+  concurrencyKey,
   onClose,
   title,
+  onUpdate,
 }: {
-  runIds?: string[];
+  concurrencyKey?: string;
   title: string | React.ReactNode;
   onClose: () => void;
+  onUpdate: () => void;
 }) => {
-  const {data} = useQuery<RunsForConcurrencyKeyQuery, RunsForConcurrencyKeyQueryVariables>(
-    RUNS_FOR_CONCURRENCY_KEY_QUERY,
+  const queryResult = useQuery<ConcurrencyKeyDetailsQuery, ConcurrencyKeyDetailsQueryVariables>(
+    CONCURRENCY_KEY_DETAILS_QUERY,
     {
       variables: {
-        limit: RUNS_LIMIT,
-        filter: {
-          runIds: runIds || [],
-        },
+        concurrencyKey: concurrencyKey || '',
       },
-      skip: !runIds || !runIds.length,
+      skip: !concurrencyKey,
     },
   );
-
-  const [freeSlots] = useMutation<
-    FreeConcurrencySlotsForRunMutation,
-    FreeConcurrencySlotsForRunMutationVariables
-  >(FREE_CONCURRENCY_SLOTS_FOR_RUN_MUTATION);
-
-  const freeSlotsActionMenuItem = React.useCallback(
-    (run: RunTableRunFragment) => {
-      return doneStatuses.has(run.status)
-        ? [
-            <MenuItem
-              key="free-concurrency-slots"
-              icon="status"
-              text="Free concurrency slots for run"
-              onClick={async () => {
-                const resp = await freeSlots({variables: {runId: run.id}});
-                if (resp.data?.freeConcurrencySlotsForRun) {
-                  await showSharedToaster({
-                    intent: 'success',
-                    icon: 'copy_to_clipboard_done',
-                    message: 'Freed concurrency slots',
-                  });
-                }
-                onClose();
-              }}
-            />,
-          ]
-        : [];
-    },
-    [freeSlots, onClose],
-  );
+  useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
+  const {data} = queryResult;
+  const refetch = React.useCallback(() => {
+    queryResult.refetch();
+    onUpdate();
+  }, [queryResult, onUpdate]);
 
   return (
     <Dialog
-      isOpen={!!runIds && runIds.length > 0}
+      isOpen={!!concurrencyKey}
       title={title}
       onClose={onClose}
-      style={{minWidth: '400px', maxWidth: 'calc(100% - 40px)', width: 'fit-content'}}
+      style={{
+        minWidth: '400px',
+        maxWidth: 'calc(100% - 40px)',
```

Comment:
Maybe just 1000px here or something instead? We shouldn't really have dialogs that occupy the full width of a super-wide viewport.